.happy/terraform/modules/ecs-stack/main.tf

```diff
@@ -180,31 +182,35 @@ module wmg_batch {
 }
 
 module upload_success_lambda {
-  source                = "../lambda"
-  image                 = "${local.lambda_upload_success_repo}:${local.image_tag}"
-  name                  = "upload-success"
-  custom_stack_name     = local.custom_stack_name
-  remote_dev_prefix     = local.remote_dev_prefix
-  deployment_stage      = local.deployment_stage
-  artifact_bucket       = local.artifact_bucket
-  cellxgene_bucket      = local.cellxgene_bucket
-  lambda_execution_role = local.lambda_execution_role
-  subnets               = local.subnets
-  security_groups       = local.security_groups
+  source                     = "../lambda"
+  image                      = "${local.lambda_upload_success_repo}:${local.image_tag}"
+  name                       = "upload-success"
+  custom_stack_name          = local.custom_stack_name
+  remote_dev_prefix          = local.remote_dev_prefix
+  deployment_stage           = local.deployment_stage
+  artifact_bucket            = local.artifact_bucket
+  cellxgene_bucket           = local.cellxgene_bucket
+  dataset_submissions_bucket = local.dataset_submissions_bucket
+  lambda_execution_role      = local.lambda_execution_role
+  step_function_arn          = "" # ugly hack to avoid circular dependency
```

Comment:
Is there a cleaner way to avoid this circular dependency? The step function needs the success/error handler lambdas and the dataset submission lambda needs the step function.

Reply:
Two options:
1. Make `step_function_arn` an optional argument in the lambda module
2. Create two lambdas submodules, one for success/failure and one for uploads

Since the lambda module is not general but depends on the lambda's business logic, arguably 2 is cleaner. 1 is much faster though